lib/board.rb

```diff
@@ -0,0 +1,73 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    horiz_line = "-----------"
+    puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts horiz_line
+    puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts horiz_line
+    puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+
+  # Accepts position as a player-perspective string (1-9)
+  def position(position)
+    index = position.to_i - 1
+    self.cells[index]
+  end
+
+  # Accepts position as a player-perspective string (1-9)
+  def update(position, player)
+    # put player.token in the correct position
+    self.cells[position.to_i - 1] = player.token
+  end
+
+  def full?
+    self.cells.all? do |element|
+      element != " " && element != "" && element != nil
+    end
+  end
+
+  def turn_count
+    counter = 0
+    self.cells.each do |element|
+      counter += 1 if element != " " && element != "" && element != nil
+    end
+    counter
+  end
+
+  # Accepts a player-perspective position as a string (1-9).  Casts it .to_i internally
+  # optionally, can be called with perspective = "array perspective"
+  def taken?(position, perspective = "player perspective")
+    if perspective == "array perspective"
+      current = self.cells[position]
+    elsif perspective == "player perspective"
+      current = self.cells[position.to_i - 1]
+    end
+    return true if current != " " && current != ""
```

Comment:
Avoid returning boolean literals from `if` statements, relying instead on the implicit return value of the boolean expression.

```
def taken?(position, perspective = "player perspective")
  ...
  current != " " && current != ""
end
```